### ISSUE TITLE
Bucket Pattern

### ISSUE THREAD

**Issue by raoulvdberge** [refinedmods/refinedstorage2#168]
A pattern that can be created with a filled bucket + pattern to ease fluid automation.

**Comment by bulldog98** [comment]
Do I understand that correctly that this pattern should take a bucket of fluid (as fluid in the system) + a bucket and produce an bucket of fluid as item?

**Comment by raoulvdberge** [comment]
> Do I understand that correctly that this pattern should take a bucket of fluid (as fluid in the system) + a bucket and produce an bucket of fluid as item?

Yes, this pattern is an alternative to having a bucket filling machine being necessary like RS 1.

This pattern would create the fluid bucket by just extracting a regular bucket from the storage network and a fluid from the storage network.

**Comment by bulldog98** [comment]
> > Do I understand that correctly that this pattern should take a bucket of fluid (as fluid in the system) + a bucket and produce an bucket of fluid as item?
> 
> Yes, this pattern is an alternative to having a bucket filling machine being necessary like RS 1.
> 
> This pattern would create the fluid bucket by just extracting a regular bucket from the storage network and a fluid from the storage network.

That is brilliant, that was always a pain in RS1 and AE2 in mod packs.

**Comment by BooBooFastCat1** [comment]
with this addition will I need 100 buckets to make 100 cakes even though the cake recipe returns the bucket?

**Comment by youssefhabri** [comment]
I assume it won't be necessary, as for example if you only have 1 bucket, the system will reserve that one, start crafting the first cake, but stop until more buckets are available, and that will resume once the first bucket is back to the system, which might be slow, but it is the same thing that happens with RS1 when you use an external bucket filling machine.

**Comment by BooBooFastCat1** [comment]
Will this also be able to empty buckets into the system as in have a bucket of lava and craft it into lava and a bucket

**Comment by BooBooFastCat1** [comment]
Would an alternative be to have it called an upgraded pattern with a allow fluids option on the crafting recipe menu which allows the fluid that is in the bucket to be used instead of the bucket as it would remove the need for extra return items of the bucket

**Comment by Diegovsky** [comment]
The system could reserve the amount of buckets required to craft one item, and if such amount exists, act as if there are enough buckets to craft the whole request.

E.g: I want to craft 100x Fooses, and one Foo needs one water bucket. RS2 would reserve one bucket and act as if it had 100 water buckets.

**Comment by aidan023** [comment]
This but being able to use fluids straight from the system to replace fluid buckets.

**Comment by raoulvdberge** [comment]
I'm not entirely sure what to do with issue - a bucket pattern wouldn't bypass reusable item issues (you'd still craft a bazillion buckets.)

Probably going to close this one.
